### PR TITLE
Migration to Robot testing framework 2

### PR DESCRIPTION
**Pull request by Nicogene**

### CODE REVIEW

File: cmake/modules/RobotTestingFrameworkTestHelpers.cmake [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+endmacro()
+
+function(ADD_CPPTEST)
+  # set(options )
+  set(oneValueArgs NAME PARAM)
+  set(multiValueArgs SRCS LIBS)
+  cmake_parse_arguments(ADD_CPPTEST "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+  set(TEST_LIBS RobotTestingFramework::RTF)
+  if(ENABLE_PLUGIN)
```
Comment (traversaro) [inline cmake/modules/RobotTestingFrameworkTestHelpers.cmake]: All these variables seems not to be defined in this project.

File: tests/librFSM/CMakeLists.txt [hunk 2]
```diff
@@ -6,7 +6,7 @@
...
 
 # LoadrFSM
-ADD_RTF_CPPTEST(NAME LoadrFSMTest
-                SRCS loadrFSM.cpp
-                PARAM "${CMAKE_SOURCE_DIR}/tests/fsm/simple_fsm.lua")
-
+include_directories(../../librFSM/include)
+ADD_RobotTestingFramework_CPPTEST(NAME LoadrFSMTest
+                                  SRCS loadrFSM.cpp
+                                  PARAM "${CMAKE_SOURCE_DIR}/tests/fsm/simple_fsm.lua")
```
Comment (traversaro) [inline tests/librFSM/CMakeLists.txt]: How is this working if rFSM is not linked?

Comment (Nicogene) [reply tests/librFSM/CMakeLists.txt]: I asked to myself the same thing :thinking: 
It used to work, let's see if it still work :sweat_smile:

Comment (traversaro) [reply tests/librFSM/CMakeLists.txt]: I suspect it is using some header-only part of the library.

Comment (Nicogene) [reply tests/librFSM/CMakeLists.txt]: It didn't work actually, it gave linking problem at runtime

**Pull request merged**